### PR TITLE
media-libs/avidemux-core: Update to EAPI 6 and revbump to 2.6.20

### PR DESCRIPTION
**Pull request by Peter-Levine**
Update to EAPI 6 and revbump to 2.6.20.

Note: Also subsumes PR https://github.com/gentoo/gentoo/pull/4797 (Fix building with GCC-6).

### CODE REVIEW

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+	!system-ffmpeg? ( dev-lang/yasm[nls=] )
+"
+
+S="${WORKDIR}/${MY_P}"
+BUILD_DIR="${S}/buildCore"
+CMAKE_USE_DIR="${S}/${PN/-/_}"
+
+DOCS=( AUTHORS README )
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: don't need these, part of the default set

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+"
+DEPEND="
+	$DEPEND
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+	!system-ffmpeg? ( dev-lang/yasm[nls=] )
+"
+
+S="${WORKDIR}/${MY_P}"
+BUILD_DIR="${S}/buildCore"
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: do you really need this? it's better to not have an explicit `BUILD_DIR` and rather fix the build system instead

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		-DSDL="$(usex sdl)"
+		-DLIBVA="$(usex vaapi)"
+		-DVDPAU="$(usex vdpau)"
+		-DXVBA="$(usex video_cards_fglrx)"
+		-DXVIDEO="$(usex xv)"
+		-DNVENC="$(usex nvenc)"
+	)
+
+	if use debug ; then
+		mycmakeargs+=( -DVERBOSE=1 -DCMAKE_BUILD_TYPE=Debug -DADM_DEBUG=1 )
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: Adding `-DCMAKE_BUILD_TYPE=Debug` in general is dodgy as cmake tends to add its own crap to `CXXFLAGS`

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 4]
```diff
@@ -0,0 +1,113 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit cmake-utils
+
+SLOT="2.6"
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: `SLOT` canonically goes after `LICENSE`

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+DESCRIPTION="Core libraries for a video editor designed for simple cutting, filtering and encoding tasks"
+HOMEPAGE="http://fixounet.free.fr/avidemux"
+
+# Multiple licenses because of all the bundled stuff.
+LICENSE="GPL-1 GPL-2 MIT PSF-2 public-domain"
+IUSE="debug nls nvenc sdl system-ffmpeg vaapi vdpau video_cards_fglrx xv"
+KEYWORDS="~amd64 ~x86"
+
+if [[ ${PV} == *9999* ]] ; then
+	KEYWORDS=""
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: remove this completely (you only ever want one `KEYWORDS=` line in an ebuild for `ekeyword` not to go nuts)

File: media-libs/avidemux-core/avidemux-core-2.6.20.ebuild [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+			|| die "Failed to avoid existing avidemux installation from making the build fail."
+	fi
+
+	# Add lax vector typing for PowerPC.
+	if use ppc || use ppc64 ; then
+		append-cflags -flax-vector-conversions
+	fi
+
+	# See bug 432322.
+	use x86 && replace-flags -O0 -O1
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.20.ebuild]: the `use ppc/ppc64` + `use x86` blocks should go in `src_configure` (pure flag editing should always happen in `src_configure`)

File: media-libs/avidemux-core/avidemux-core-2.6.8.ebuild [hunk 7]
```diff
@@ -84,18 +85,18 @@ src_prepare() {
...
+	local mycmakeargs=(
 		-DAVIDEMUX_SOURCE_DIR='${S}'
 		$(cmake-utils_use nls GETTEXT)
 		$(cmake-utils_use sdl SDL)
 		$(cmake-utils_use vaapi LIBVA)
 		$(cmake-utils_use vdpau VDPAU)
 		$(cmake-utils_use video_cards_fglrx XVBA)
 		$(cmake-utils_use xv XVIDEO)
-	"
+	)
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-2.6.8.ebuild]: thank you :+1:

File: media-libs/avidemux-core/avidemux-core-9999-r1.ebuild [hunk 8]
```diff
@@ -0,0 +1,113 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+inherit cmake-utils
+
+SLOT="2.6"
```
Comment (SoapGentoo) [inline media-libs/avidemux-core/avidemux-core-9999-r1.ebuild]: do the same edits here as well (and don't bump the `-9999`)

**Comment by aballier** [comment]
merged; note that bumping only -core will have not much effect until you bump avidemux itself

also, it'd be great if we could finally make it use system ffmpeg instead of its bundled security hazard ffmpeg

**Pull request closed without merging**